### PR TITLE
MIR-434 Phase 1: Extract network utilities

### PR DESCRIPTION
**Pull request by phinze**
## Summary

Extract network address parsing and URL building into a centralized `pkg/netutil` package to eliminate duplication across the codebase.

**Part of MIR-434** (Activator Simplification) - [RFD 0034](https://rfd.miren.garden/rfd/34) Phase 1

## Changes

- **New package**: `pkg/netutil` with `ParseNetworkAddress()` and `BuildURL()` functions
- **Eliminates duplication**: Removes 3 copies of CIDR parsing logic from:
  - `components/activator/activator.go` (`activateApp()` and `recoverSandboxes()`)
  - `pkg/computex/sandbox.go` (`WaitForSandbox()`)

## Benefits

- Single source of truth for network address parsing
- Consistent error handling and messaging across components
- Easily testable in isolation (14 test cases covering CIDR, plain IP, IPv4/IPv6, error cases)
- Reduces activator from 918 to 908 lines

## Testing

- ✅ All netutil unit tests pass (14 test cases)
- ✅ Existing activator tests pass
- ✅ Code formatted and vetted

## Test Plan

```bash
go test miren.dev/runtime/pkg/netutil
go test miren.dev/runtime/components/activator
go test miren.dev/runtime/pkg/computex
```

## Next Steps

Phase 2 will build on this by moving config resolution to build time, further simplifying the activator.

## Summary by CodeRabbit

* **New Features**
  * More robust handling of IPv4/IPv6/CIDR inputs and consistent HTTP service URL construction.

* **Bug Fixes**
  * Fewer activation/recovery failures by standardizing parsing and propagating address-construction errors.

* **Refactor**
  * Centralized network address parsing and URL building into a shared utility used across components.

* **Tests**
  * Added unit tests for valid/invalid address parsing and URL generation, including IPv6 and edge cases.

### CODE REVIEW

File: pkg/netutil/netutil.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		return ip.String(), nil
+	}
+
+	return "", fmt.Errorf("invalid address format: %s", addr)
+}
+
+// BuildURL builds HTTP URL from IP and port
+func BuildURL(ip string, port int64) string {
+	return fmt.Sprintf("http://%s:%d", ip, port)
+}
```
[the remark this answers is not in the document]
Comment (evanphx) [reply pkg/netutil/netutil.go]: This is an important one, 🐰 is right.

**Review by evanphx — CHANGES_REQUESTED** [review]
Need to use net.JoinHostPort, otherwise looks good!

**Review by evanphx — APPROVED** [review]
